pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,318 @@
+        values=[30000, 8000000000],
+    )
+
+    span_frequency = Instrument.control(
+        ":FREQ:SPAN?",
+        ":FREQ:SPAN %d",
+        """Span Frequency in Hz""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    start_frequency = Instrument.control(
+        ":FREQ:STAR?",
+        ":FREQ:STAR %d",
+        """ Starting frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    stop_frequency = Instrument.control(
+        ":FREQ:STOP?",
+        ":FREQ:STOP %d",
+        """ Stop frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    sweep_type = Instrument.control(
+        ":SWE:TYP?",
+        ":SWE:TYP %s",
+        """ Sweep Type. Valid options are LIN|LOG|FSEGM|ISEGM|POW|MFGC.""",
+        validator=strict_discrete_set,
+        values=_SWEEP_TYPES,
+    )
+
+    sweep_mode = Instrument.control(
+        ":SA:MODE?",
+        ":SA:MODE %s",
+        """ Sweep mode. Valid options are VNA|CLAS.""",
+        validator=strict_discrete_set,
+        values=["VNA", "CLAS"],
+    )
+
+    sweep_points = Instrument.control(
+        ":SWE:POI?", ":SWE:POI %d", """ Sweep point.""", validator=strict_range, values=[2, 100000]
+    )
+
+    num_averages = Instrument.control(
+        ":AVER:COUN?",
+        ":AVER:COUN %d",
+        """ Sets the number of averages.""",
+        validator=strict_range,
+        values=[1, 20000],
+    )
+
+    bandwidth = Instrument.control(
+        ":BWID?",
+        ":BWID %d",
+        """ Sets the IF bandwidth.""",
+        validator=strict_range,
+        values=[1, 1000000],
+    )
+
+    def insert_id(self, command):
+        return f":SENS{self.id}{command}"
+
+
+class AnritsuMS464xB(Instrument):
+    """Represents the Anritsu MS474xB VNA. Implements controls to change the analysis
+    range and to retreve the data for the trace.
+    """
+
+    channels = Instrument.ChannelCreator(
+        VNAChannel, (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)
+    )
+
+    class BinaryData_Byte_Format(CustomEnum):
+        MSB_FIRST = "NORM"
+        LSB_FIRST = "SWAP"
+
+    class DataFileParameter(CustomEnum):
+        """Parameter format in a data file"""
+
+        LINEAR_AND_PHASE = "LINPH"
+        LOG_AND_PHASE = "LOGPH"
+        REAL_AND_IMAGINARY_NUMBERS = "REIM"
+
+    class BinaryDataFormat(CustomEnum):
+        """Binary Data file format."""
+
+        ASC = "ASC"
+        REAL = "REAL"
+        REAL32 = "REAL32"
+
+    class DataFileFrequency(CustomEnum):
+        """Binary Data file frequency units."""
+
+        HZ = "HZ"
+        KHZ = "KHZ"
+        MHZ = "MHZ"
+        GHZ = "GHZ"
+
+    def __init__(self, adapter, name="Anritsu MS464xB", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    def load_data_file(self, filename: str):
+        """Loads a data file in VNA HDD.
+
+        Args:
+            filename (str): full filename including path
```

Comment:
In our documentation (sphinx), we have to use a different style:
```suggestion
        :param str filename: full filename including path
```
The "param" precedes every single argument (and that "args" is superfluos).
The type may be added between the param keyword and the parameter name.

Reply:
Specifically, this is the ReST format (https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists), see https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings.